### PR TITLE
Feat(unit tests): Unit Test for enums.py

### PR DESCRIPTION
**Pull request by Physsix27**
Description:
>This is just the unit tests for the enums.py file and I want it to know if this is what you want.
> Then I can make more Pull Requests for different unit tests depending on the file that I am testing.
> I will also be including the coverage report at some point too.

Also, I added a `requirements_unittest.txt` file with a few requirements I need it when running tests in a virtual environment, I did not want to add it to other requirements.txt that you have because I am not sure if you want that or where you want it to be added, which is just the `coverage==4.5.4` dependency.

Reference Issue: #7 

### Pre-Pull Request Checklist:
- [x] Run `make test`

### CODE REVIEW

**Comment by Fogapod** [comment]
Pre-commit tests are failing. Please, fix this before I could merge

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@ __pycache__/
 htmlcov/
 .tox/
 .nox/
+.pyc
```
Comment (Fogapod) [inline .gitignore]: Why is this necessary? `__pycache__` is already included,  what produces `.pyc` files outside `__pycache__`?

File: tests/unit/enums_py/StatusCode_test.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    """
+
+    def test_enums_variables_values(self):
+        """
+        Should ensure that variables are set to their expected values
+        """
+
+        all_enumarations_values = [
+            StatusCode.SUCCESS.value,
+            StatusCode.BAD_FORMAT.value,
```
Comment (Fogapod) [inline tests/unit/enums_py/StatusCode_test.py]: Can `all_enumarations_values` be moved outside to prevent code duplication?

**Comment by Fogapod** [comment]
> This is just the unit tests for the enums.py file and I want it to know if this is what you want.
> Then I can make more Pull Requests for different unit tests depending on the file that I am testing.
> I will also be including the coverage report at some point too.

I was originally planning on using pytest for testing and coverage reports, is this possible?

> Also, I added a `requirements_unittest.txt` file with a few requirements I need it when running tests in a virtual environment, I did not want to add it to other requirements.txt that you have because I am not sure if you want that or where you want it to be added, which is just the coverage==4.5.4 dependency.

`requirements_unittest.txt` could be moved to `tests` folder.

One more thing, could you add tests to travis? I'm planning to use it to run tests.

File: .gitignore (hunk 1 above)
Comment (Physsix27) [reply .gitignore]: It is not necessary, I will remove it.

File: tests/unit/enums_py/StatusCode_test.py (hunk 2 above)
Comment (Physsix27) [reply tests/unit/enums_py/StatusCode_test.py]: I will clean my code.

**Comment by Physsix27** [comment]
Do you mean adding the test runs to the Travis CI, I am not 100% sure what you mean.

Also, I would recommend using the [unittest.mock]  python library (https://docs.python.org/3/library/unittest.mock.html#module-unittest.mock) instead of the `pytests`

**Comment by Physsix27** [comment]
Can I get some help, I have not to work with Travis before and I am not sure why it says it can't find the module or it's failing the isort and black which I don't know what they are either. Some help.

File: requirements-dev.txt [hunk 3]
```diff
@@ -18,3 +18,4 @@ six==1.12.0               # via cfgv, pip-tools, pre-commit
 toml==0.10.0              # via pre-commit
 virtualenv==16.7.5        # via pre-commit
 zipp==0.6.0               # via importlib-metadata
+coverage==4.5.4           # need it for running tests
```
Comment (Fogapod) [inline requirements-dev.txt]: This file is autogenerated and shouldn't be modified manually. Create separate file for test requirements instead (`tests/requirements.txt`)

File: requirements.txt [hunk 4]
```diff
@@ -7,3 +7,4 @@
 aioredis==1.2.0
 async-timeout==3.0.1      # via aioredis
 hiredis==1.0.0            # via aioredis
+coverage==4.5.4           # need it for running tests
```
Comment (Fogapod) [inline requirements.txt]: 1) This file is autogenerated and shouldn't be modified manually. Create separate file for test requirements instead (`tests/requirements.txt)
2) These are runtime requirements, coverage shouldn't be there

File: makefile [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	@rm -rf $(COVERAGE_DIR)
+	@rm -rf .coverage
+
+.PHONY: help
+help:
+	@echo 'List all the make tergets available for use with "make list"'
+
+.PHONY: list
+list:
+	@$(MAKE) -pRrq -f $(lastword $(MAKEFILE_LIST)) : 2>/dev/null | awk -v RS= -F: '/^# File/,/^# Finished Make data base/ {if ($$1 !~ "^[#.]") {print $$1}}' | sort | egrep -v -e '^[^[:alnum:]]' -e '^$@$$'
```
Comment (Fogapod) [inline makefile]: Can you please explain this line?

**Comment by Fogapod** [comment]
> Do you mean adding the test runs to the Travis CI, I am not 100% sure what you mean.

Yes
> Also, I would recommend using the unittest.mock python library instead of the `pytests`

pytest works with unittest.mock perfectly fine
